internal/internal_worker_base.go

```diff
@@ -401,3 +410,38 @@ func (bw *baseWorker) Stop() {
 	}
 	return
 }
+
+func (bw *baseWorker) EmitHardwareUsage() {
+	defer bw.shutdownWG.Done()
+
+	emitOnce.Do(func() {
```

Comment:
Mind adding a comment as to the sync.Once ussage? I am guessing it's because you don't want duplicated metrics but that's kinda non-obvious to the casual reader

Reply:
Done